test/main_test.go

```diff
@@ -33,12 +33,22 @@ import (
 )
 
 const (
-	disabledTag    = "@disabled"
-	cliTag         = "@cli"
-	smokeTag       = "@smoke"
-	performanceTag = "@performance"
+	disabledTag = "@disabled"
+	cliTag      = "@cli"
 )
 
+var customTags = []customTag{
+	{Name: "@performance", IsEnabled: config.IsPerformanceTests, IgnoreIfDisabled: true},
+	{Name: "@release", IsEnabled: config.IsReleaseTests, IgnoreIfDisabled: true},
+	{Name: "@smoke", IsEnabled: config.IsSmokeTests, IgnoreIfDisabled: false},
+}
+
+type customTag struct {
+	Name             string
+	IsEnabled        func() bool
+	IgnoreIfDisabled bool
```

Comment:
This might be tricky as it's only used by the smoke tag because it's not excluded if it's not set. If smoke tag is set, we only run the smoke tests, if not set, we run all the tests (including smokes). 
If this is too much, I can restore it as it was before, but I like the idea of having all the custom tag configurations in the same place to see/change behaviour when/if needed.

Reply:
I think both are tricky and a bit verbose. More ideas? IgnoreByDefault?